iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -342,6 +346,32 @@ private void Validate()
                 throw new ArgumentException(
                     "Should not specify either SharedAccessKey or SharedAccessSignature if X.509 certificate is used for authenticating the client with IoT hub.");
             }
+
+            // Validate certs.
+            if (AuthenticationMethod is DeviceAuthenticationWithX509Certificate)
+            {
+                // Prep for certificate auth.
+                if (Certificate == null)
+                {
+                    throw new ArgumentException("No certificate was found. To use certificate authentication certificate must be present.");
```

Comment:
When throwing ArgumentException and ArgumentNullException, let's be sure to specify the parameter name.
```suggestion
                    throw new ArgumentException("No certificate was found. To use certificate authentication certificate must be present.", nameof(authenticationMethod));
```